packages/components/src/form-toggle/stories/index.js

```diff
@@ -0,0 +1,25 @@
+/**
+ * WordPress dependencies
+ */
+import { useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import FormToggle from '../';
+
+export default { title: 'Form Toggle', component: FormToggle };
+
+export const _default = () => {
+	const [ isChecked, setChecked ] = useState( true );
```

Comment:
The same comment as in https://github.com/WordPress/gutenberg/pull/18070#discussion_r339419618:

> According to documentation, the function exported as a story isn't a React component. It means that we shouldn't use hooks in this context as they might stop to work in the near future if they change the way all that works.
> 
> The easiest way to get rid of this warning is to move the part which manages state out of this function. See a similar usage:
> https://github.com/WordPress/gutenberg/blob/a6e1252886e8d990627ac4bf4796f768bfad2a44/packages/components/src/checkbox-control/stories/index.js#L18-L28

Related docs section: https://storybook.js.org/docs/formats/component-story-format/#story-exports

> By default every named export in the file represents a story function.
> 
> Story functions can be annotated with a story object to define story-level decorators and parameters, and also to define the name of the story.

If you would name this export differently, lint rules related to React hooks would warn as well, as they enforce to be inside a React component (the name needs to start with uppercase).

@ItsJonQ @mkaz - do you think we should be more strict about it? I know that the StoryShots integration I explored will error when you use React hooks in story functions.

Reply:
@gziolo After creating a couple of stories, I started doing something like this:

```jsx
...

const Example = () => {
	const [ isChecked, setChecked ] = useState( true );
	return (
		<FormToggle
			checked={ isChecked }
			onChange={
				() => {
					setChecked( ! isChecked );
				}
			}
		/>
	);
}

export const _default = () => {
	return <Example />
};
```

This bypasses the React Hook issue that StoryShots/Book complains about.

What do you think @enejb ? :D